test/L1/SwapAndBridge.t.sol

```diff
@@ -0,0 +1,169 @@
+contract TestSwapAndBridge is Test {
+    // Address used for unit tests
+    address constant testAccount = address(0xc0ffee);
+
+    // test account balance
+    uint256 testBalance = 1e60;
+
+    function setUp() public {
+        vm.deal(testAccount, testBalance);
+    }
+
+    function test_InvalidConstructor() public {
+        vm.expectRevert("Invalid L1 bridge address.");
+        new SwapAndBridge(address(0), address(0xdead), address(0xbeef));
+
+        vm.expectRevert("Invalid L1 token address.");
+        new SwapAndBridge(address(0xbeef), address(0), address(0xdead));
+
+        vm.expectRevert("Invalid L2 token address.");
+        new SwapAndBridge(address(0xbeef), address(0xdead), address(0));
+    }
+
+    function test_InvalidRecipient() public {
+        SwapAndBridge swapAndBridge = new SwapAndBridge(address(0xdead), address(0xbeef), address(0x1111));
+        vm.expectRevert("Invalid recipient address.");
+        swapAndBridge.swapAndBridgeTo{ value: 1 }(address(0));
+    }
+
+    function test_ReceiveFallback(uint256 tokensPerETH, uint256 value) public {
+        // We bound tokensPerETH in fuzzzing to avoid overflows in the MockERC20LST contract
+        // and 'No wrapped tokens minted.' and 'Invalid msg value' reverts.
+        vm.assume(tokensPerETH > 0);
+        vm.assume(tokensPerETH <= 1e38);
+        vm.assume(value > 0);
+        vm.assume(value <= 1e38);
+        // Ensure that some tokens will be minted in the MockERC20LST contract.
+        vm.assume(tokensPerETH * value >= 1e18);
+
+        MockERC20LST mockLST = new MockERC20LST(tokensPerETH);
+        MockStandardBridge bridge = new MockStandardBridge();
+        SwapAndBridge swapAndBridge = new SwapAndBridge(address(bridge), address(mockLST), address(0xbeef));
+
+        vm.startPrank(testAccount);
+        (bool sent,) = address(swapAndBridge).call{ value: value }("");
+        require(sent == true, "Failed to send Ether.");
+        vm.stopPrank();
+    }
+
+    function test_SwapAndBridgeTo(uint256 tokensPerETH, uint256 value, address recipient) public {
+        // We bound tokensPerETH in fuzzzing to avoid overflows in the MockERC20LST contract
+        // and 'No wrapped tokens minted.' and 'Invalid msg value' reverts.
+        vm.assume(tokensPerETH > 0);
+        vm.assume(tokensPerETH <= 1e38);
+        vm.assume(value > 0);
+        vm.assume(value <= 1e38);
+        // Ensure that some tokens will be minted in the MockERC20LST contract.
+        vm.assume(tokensPerETH * value >= 1e18);
+
+        MockERC20LST mockLST = new MockERC20LST(tokensPerETH);
+        MockStandardBridge bridge = new MockStandardBridge();
+        SwapAndBridge swapAndBridge = new SwapAndBridge(address(bridge), address(mockLST), address(0xbeef));
+
+        vm.startPrank(testAccount);
+        if (recipient != address(0)) {
+            swapAndBridge.swapAndBridgeTo{ value: value }(recipient);
+        }
+        vm.stopPrank();
```

Comment:
We tested that `swapAndBridgeTo` function doesn't revert.
May we also check below if some other conditions are true like `testAccount balance (on L1, possibly also on L2?)`, `bridge balance`, etc.?
So that state after `swapAndBridgeTo` is executed is correct?

Reply:
Same is true for other test functions below except the ones which test that `revert` occurs.

Reply:
testing the balance on the L2 is not possible unless we run an E2E script, as it requires also dispatching the message to the L2 node, but I think this is covered in the e2e tests. I added checks for the L1 state whenever possible

Reply:
Ok, then it's fine how it is implemented.